### PR TITLE
refactor(link-preview): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Converts enzyme tests to RTL

### Current behaviour

Currently, tests are written in enzyme

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/link-preview/link-preview.test.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  render(<LinkPreview url="https://foo" />);
+  const linkText = screen.getByText("foo");
+
+  expect(linkText).toBeVisible();
+});
+
+test('renders with a div root element, when the `as` prop is passed as "div"', () => {
+  render(<LinkPreview as="div" />);
+  const linkPreviewWrapper = screen.getByTestId("link-preview");
+  expect(linkPreviewWrapper.tagName).toBe("DIV");
```
Comment (DipperTheDan) [inline src/components/link-preview/link-preview.test.tsx:31]: **suggestion/nitpick:** The first three tests have a line break before the `expect` assertion. I'm happy with either but make it consistent.

File: src/components/link-preview/link-preview.test.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import LinkPreview from "./link-preview.component";
+
+test("renders with an anchor root element and hrf when the `url` prop is passed", () => {
```
Comment (DipperTheDan) [inline src/components/link-preview/link-preview.test.tsx:6]:
```suggestion
test("renders with an anchor root element and href when the `url` prop is passed", () => {
```

**Review by nuria1110 — COMMENTED** [review]
Was going to point out how it may be good to also have a test for the default 'alt' text for an image if just the 'url' is passed and another to assert the `Placeholder` is rendered instead if an image isn't provided, but also wondering how these branches are getting covered in the first place 🤔 

Apart from Dan's comments the rest looks good to me 👍

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**